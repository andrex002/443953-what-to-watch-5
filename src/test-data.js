export const film = {
  "id": 1,
  "title": "The Grand Budapest Hotel",
  "image": "img/the-grand-budapest-hotel-poster.jpg",
  "previewImage": "img/the-grand-budapest-hotel.jpg",
  "bgImage": "img/the-grand-budapest-hotel-bg.jpg",
  "bgColor": "#ffffff",
  "srcVideo": "https://some-link",
  "video": "https://some-link",
  "previewVideo": "https://some-link",
  "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.",
  "rating": {
    "score": 8.9,
    "count": 240
  },
  "director": "Wes Andreson",
  "actors": ["Bill Murray", "Edward Norton", "Jude Law", "Willem Dafoe", "Saoirse Ronan"],
  "duration": 99,
  "genre": "Comedy",
  "year": 2014,
  "isFavorite": false
};

export const films = [
  {
    "id": 1,
    "title": "The Grand Budapest Hotel",
    "image": "img/the-grand-budapest-hotel-poster.jpg",
    "previewImage": "img/the-grand-budapest-hotel.jpg",
    "bgImage": "img/the-grand-budapest-hotel-bg.jpg",
    "bgColor": "#ffffff",
    "video": "https://some-link",
    "previewVideo": "https://some-link",
    "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.",
    "rating": {
      "score": 8.9,
      "count": 240
    },
    "director": "Wes Andreson",
    "actors": ["Bill Murray", "Edward Norton", "Jude Law", "Willem Dafoe", "Saoirse Ronan"],
    "duration": 99,
    "genre": "Comedy",
    "year": 2014,
    "isFavorite": false
  },
  {
    "id": 2,
    "title": "The Grand Budapest Hotel",
    "image": "img/the-grand-budapest-hotel-poster.jpg",
    "previewImage": "img/the-grand-budapest-hotel.jpg",
    "bgImage": "img/the-grand-budapest-hotel-bg.jpg",
    "bgColor": "#ffffff",
    "video": "https://some-link",
    "previewVideo": "https://some-link",
    "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.",
    "rating": {
      "score": 8.9,
      "count": 240
    },
    "director": "Wes Andreson",
    "actors": ["Bill Murray", "Edward Norton", "Jude Law", "Willem Dafoe", "Saoirse Ronan"],
    "duration": 99,
    "genre": "Comedy",
    "year": 2014,
    "isFavorite": false
  }
];

export const comment = {
  "id": 1,
  "userId": 4,
  "userName": "Kate Muir",
  "rating": 8.9,
  "textComment": "Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.",
  "date": "2019-05-08T14:13:56.569Z"
};

export const comments = [
  {
    "id": 1,
    "userId": 4,
    "userName": "Kate Muir",
    "rating": 8.9,
    "textComment": "Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.",
    "date": "2019-05-08T14:13:56.569Z"
  },
  {
    "id": 2,
    "userId": 4,
    "userName": "Kate Muir",
    "rating": 8.9,
    "textComment": "Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.",
    "date": "2019-05-08T14:13:56.569Z"
  },
];

export const defaultState = {
  DATA: {
    allFilms: films,
    filteredFilms: films,
    promoFilm: film,
    currentFilm: film,
    favoriteFilms: films,
    comments: comments,
    numberFilmsShown: 8,
    isCommentSending: false,
    isCommentSendError: false
  },
  FILTER: {
    activeGenre: `All genres`,
  },
  USER: {
    authorizationStatus: `NO_AUTH`,
    name: `userName`,
    avatar: `avatar.jpg`
  }
};

export const noop = () => {};