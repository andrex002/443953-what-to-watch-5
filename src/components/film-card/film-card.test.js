import React from "react";
import renderer from "react-test-renderer";
import FilmCard from "./film-card";
import {BrowserRouter} from "react-router-dom";
import {films, noop} from "../../test-data";

const {image, title, srcVideo} = films[0];

describe(`Should FilmCard render correctly`, () => {
  it(`With active`, () => {
    const tree = renderer.create(
      <BrowserRouter>
        <FilmCard
          image={image}
          title={title}
          id={1}
          srcVideo={srcVideo}
          isActive={true}
          handleHoverFilmCard={noop}
          onFilmCardClick={noop}
        />
      </BrowserRouter>,
      {
        createNodeMock: () => {
          return {};
        }
      }
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`With no active`, () => {
    const tree = renderer.create(
      <BrowserRouter>
        <FilmCard
          image={image}
          title={title}
          id={1}
          srcVideo={srcVideo}
          isActive={false}
          handleHoverFilmCard={noop}
          onFilmCardClick={noop}
        />
      </BrowserRouter>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
