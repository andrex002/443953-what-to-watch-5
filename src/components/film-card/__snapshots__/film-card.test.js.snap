// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should FilmCard render correctly With active 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  data-id={1}
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <video
    height="175"
    muted={true}
    poster="img/the-grand-budapest-hotel-poster.jpg"
    src="https://some-link"
    width="280"
  />
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="/films/1"
      onClick={[Function]}
    >
      The Grand Budapest Hotel
    </a>
  </h3>
</article>
`;

exports[`Should FilmCard render correctly With no active 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  data-id={1}
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="small-movie-card__image"
  >
    <img
      alt="The Grand Budapest Hotel"
      height="175"
      src="img/the-grand-budapest-hotel-poster.jpg"
      width="280"
    />
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="/films/1"
      onClick={[Function]}
    >
      The Grand Budapest Hotel
    </a>
  </h3>
</article>
`;
